### DISCUSSION TITLE
Network Issue in Github Codespaces

### DISCUSSION THREAD

**Discussion by v0dev** [community/community#124964] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hi,
I have been using Codespaces for months in Firefox (always private mode) till now, last week I started getting issue while opening codespaces , i started getting Network_error , it remained like that for few days and was unable to use on Firefox (using the 126.0), the issue got resolved yesterday and I was able to use it again on Firefox, but today again getting the same error while it works on other browser (Brave, in privateMode) .

The issue for this Network Error , seems to be a PUT network request to `https://asse.rel.tunnels.api.visualstudio.com/tunnels/{id}/ports/16635?api-version=2023-09-27-preview`

I have removed the {id} from url as unsure if was sensitive, but it is similar to the codespace naming (word-word-7chars) , and it is not the same as the codespace name and is different for each code space. 

In the network tab it is shown as blocked, and in console it shows:
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://asse.rel.tunnels.api.visualstudio.com/tunnels/{id}/ports/16635?api-version=2023-09-27-preview. (Reason: CORS request did not succeed). Status code: (null).`

**Comment by Seazartea** [comment]
I believe I've been having the same issue with trying to open a codespace while using Firefox browser that started on the 14th May with it giving a Network error screen, I was using the codespace at the time it first happened when it disconnected me and trying to log back in immediately gave me the Network error screen.

The codespace starts to load and can see the outlines loaded in, as below: 
![Screenshot 2024-05-20 at 18-50-36 https __studious-parakeet-5ggv674r6p69h469w github dev](https://github.com/community/community/assets/150829409/558d8da0-e2f8-4437-868f-6888cc1b9bb2)


Then the progress bar in the bottom right corner starts to slow down halfway to almost a crawl, then speeds up before hitting me with the network error screen:
![Screenshot 2024-05-15 at 16-44-12 Welcome - 150829409 - Visual Studio Code](https://github.com/community/community/assets/150829409/2888016a-69f8-4856-b3ae-b6ae413dfe30)

I haven't figured out at all what's caused this or how to figure it out as I'm new to coding and using codespaces, but believe it's likely the same issue as it seems like it first started to happen at the same time and using the same browser. I've been able to access the codespace on another browser so know it's not due to the network I'm using or internet stability.

**Reply by v0dev** [reply]
Yes, its the same issue it seems as it acts the same way as you sent in screenshot, as soon as the request goes to the URL (which i mentioned), you get the screen in the 2nd screenshot. 
Yes the issue is not network, as works perfectly fine in other browsers, it could be due to something with Firefox but again it worked for a day in between (there has been no change in firefox or configs since the issue started) since the issue came so it might be either with Firefox or with the domain itself.
Do you use private window on Firefox, or is it occurring for you in normal window as well?

**Reply by Seazartea** [reply]
I just use the normal Firefox browser, not the private browser for accessing the codespace

**Reply by Avasara** [reply]
I'm having the same trouble as well. `https://use.rel.tunnels.api.visualstudio.com/tunnels/{ID}/ports/16635?api-version=2023-09-27-preview `is the error I'm getting. There is no status code for it too. It just shows a crossed-out circle. I'm also using Firefox. I've updated everything and stopped all extensions from running on the page (it worked fine with extensions blocking some stuff out a few hours ago though). My network also works fine.

**Reply by guyluz11** [reply]
When I try to open codespace I get an error.
The repo is open: Repo: https://github.com/guyluz11/infinite_horizons 

I have tried with 2 different users and in different browsers.

I tested the codespace from a different repo and it opened for me.
But in this one I don't even arrive to the IDE

![image](https://github.com/community/community/assets/9304740/9a1daed3-a5cd-4df8-ba77-80cd84186b81)

**Reply by kaspar-allenbach** [reply]
I have the same issue. Tried in Multiple WLANs and Computers. Works under Edge Browser . Doesn't work in Firefox no matter Windows or macOS

**Reply by kaspar-allenbach** [reply]
I found a solution. At least for Firefox. You have to disable security settings.
Click on the little shield in the adress bar and exclude security setting for *.github.dev then it works

**Reply by Alok-112** [reply]
Thanks you bro now it working fine 
I Think Github Codespaces uses some tunneling process to connect to the virtual machine 
this shield stops it from connecting by implementing firewall

**Reply by mjbear** [reply]
Based on kaspar-allenbach's comment above, I clicked the shield icon, then disabled **Enhanced Tracking Protection** (for just this github.dev site) by clicking the slider button.

github.dev loads my Codespace now!
(now I have to remember about this to another time :wink:)

**Comment by sayhiben** [comment]
My organization is also encountering frequent interruptions from disconnects, though we don't receive the error shown in the photos

**Comment by WaterSwallow** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> Hi, I have been using Codespaces for months in Firefox (always private mode) till now, last week I started getting issue while opening codespaces , i started getting Network_error , it remained like that for few days and was unable to use on Firefox (using the 126.0), the issue got resolved yesterday and I was able to use it again on Firefox, but today again getting the same error while it works on other browser (Brave, in privateMode) .
> 
> The issue for this Network Error , seems to be a PUT network request to `https://asse.rel.tunnels.api.visualstudio.com/tunnels/{id}/ports/16635?api-version=2023-09-27-preview`
> 
> I have removed the {id} from url as unsure if was sensitive, but it is similar to the codespace naming (word-word-7chars) , and it is not the same as the codespace name and is different for each code space.
> 
> In the network tab it is shown as blocked, and in console it shows: `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://asse.rel.tunnels.api.visualstudio.com/tunnels/{id}/ports/16635?api-version=2023-09-27-preview. (Reason: CORS request did not succeed). Status code: (null).`

It seems like you are having a recurring issue with GitHub Codespaces when using Firefox in private mode, and it's related to a network error involving a blocked PUT request.

The error message "Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource..." is a security feature in web browsers known as Same-Origin Policy. It prevents a script from one origin from interacting with resources from a different origin. This is a measure against cross-site scripting attacks (XSS).

This could be happening due to several reasons such as browser settings (particularly since it is specifically occurring in Firefox Private mode), firewall settings, network restrictions, or even certain browser extensions that might interfere with cross-origin requests.

Given the fact that the issue is intermittent and that the problem does not persist when using another browser (like Brave in private mode), it suggests that the issue is likely on the client-side, specifically with the Firefox settings or extensions.

You could try to troubleshoot this issue by:

- Temporarily disabling all Firefox extensions to see if any of them may be causing the issue.
- Checking if you have enabled 'Enhanced Tracking Protection' in Firefox. This feature can sometimes block necessary cookies or scripts.
- Trying to reproduce the issue in Firefox without private mode to see if there is something specific about the private browsing settings causing this issue.

If the problem persists, it would be best to reach out directly to GitHub support, providing all the details and steps you've tried so far. They may be able to provide more insights or escalate this to the engineering team if it's a bug related to GitHub Codespaces with Firefox.

**Comment by 54145a** [comment]
Same issue, can someone take a look at this, it has been like this for a few months.

**Comment by watchingdogs** [comment, marked as answer]
Disabling the tracking protection solved this issue for me as well.
However, I still don't have code highlighting, for which I'll be posting another thread.

**Reply by v0dev** [reply]
thanks, this works , but isn't really good enough solution long term, hope firefox/github fixes it

**Reply by 54145a** [reply]
What if my firewall is countrywide😭

**Reply by watchingdogs** [reply]
> What if my firewall is countrywide😭

Then you wouldn't be able to even access the page. BTW you can just spin up a free Amazon EC2 instance anytime to run a VPN.
